### DISCUSSION TITLE
How to gain full control over pluralization?

### DISCUSSION THREAD

**Discussion by valoricDe** [intlify/vue-i18n#1657] [Q&A]
Previously in version 8 I was able to overwrite the prototype function fetchChoice

```
VueI18n.prototype.fetchChoice = function fetchChoice(message, choice) { handle things }
```
message in my case is an object and not a string. That's why the current public function to change the choice making is not enough for me.

But with version 9 and createI18n function I do not know how to do it.
I saw the referenced line (https://github.com/intlify/vue-i18n-next/blob/b5c73ef661e88719cb5d6c40a58e264923230077/packages/core-base/src/runtime.ts#L409C5-L409C36): `[HelperNameMap.PLURAL]: plural` in `createMessageContext`.
And I wondered how I can overwrite the Helper plural.

**Comment by ChrisGV04** [comment]
Hi! I found that the v9 documentation has a section for custom pluralization. Maybe this can help you:
https://vue-i18n.intlify.dev/guide/essentials/pluralization.html#custom-pluralization

**Reply by valoricDe** [reply]
Unfortunately not. Because it only allows to return the choice but you have no access to the whole object.

So let's say your translation would be: `{ one: "banana", two: "bananas" ... }` instead of `banana|bananas|bananana` then you have no way to pluralize this in the new implementation 🤷

**Reply by ChrisGV04** [reply]
Oh I see what you mean. In that case I don't know what could be done...

Unfortunately I have struggled a lot with this library compared to the more popular `i18next`.

Hopefully someone with more in-depth knowledge of the library can assist you soon!
